### PR TITLE
Add OPC water model

### PR DESCRIPTION
**Pull request by chapincavender**
This PR adds a SMIRNOFF port of the OPC water model and ions parameters from AmberTools 22.

The nonbonded energies for a box of OPC water and sodium chloride match to at least eight digits of precision:
```
Term                 System  Energy (kJ mol^-1)
Total Nonbonded      OpenMM   -63658.63164652
Total Nonbonded      OpenFF   -63658.63164652
Coulomb              OpenMM   -73747.77246641
Coulomb              OpenFF   -73747.77246641
Lennard-Jones        OpenMM    10089.14081989
Lennard-Jones        OpenFF    10089.14081989
Coulomb Water-Water  OpenMM   -54595.00449227
Coulomb Water-Water  OpenFF   -54595.00449227
Coulomb Ion-Ion      OpenMM    -4314.50032548
Coulomb Ion-Ion      OpenFF    -4314.50032548
Coulomb Water-Ion    OpenMM   -14838.26764866
Coulomb Water-Ion    OpenFF   -14838.26764866
Coulomb Non-Virtual  OpenMM   223876.03756013
Coulomb Non-Virtual  OpenFF   223876.03756013
Coulomb Virtual      OpenMM   162803.17266965
Coulomb Virtual      OpenFF   162803.17266965
Coulomb VS-NVS       OpenMM  -460426.98269619
Coulomb VS-NVS       OpenFF  -460426.98269619
```

TODOs:
* [ ] Add tests in this repo
* [ ] Update top level README

### CODE REVIEW

**Review by j-wags — APPROVED** [review]
Approved pending formatting fixes (which I'll go ahead and merge right now)

File: openforcefields/tests/test_water_models.py
Comment (mattwthompson) [inline openforcefields/tests/test_water_models.py:95]: Any suspicion if this is a problem with the force field or Interchange?

Comment (j-wags) [reply openforcefields/tests/test_water_models.py:95]: We squinted a lot at the force field and couldn't spot a problem there, but it's a very confusing issue. The things we figured out are:

* The atom/particle indexing are the same between the OpenMM-generated system and the OpenFF-generated one
* The disagreeing sigmas are `0.890898` and `1.78179743628`, which are `1 / (2^(1/6))` and `2 / (2^(1/6))` respectively. 

We didn't see any mention of `rmin_half=2. * unit.angstrom`, which would seem to be the only way we should end up with the latter. So it seems like it may be an interchange bug, maybe having to do with the multiplicity of matches in the vsites. @lilyminium is working on a minimal reproducing example and will open an issue when she has one.

**Comment by lilyminium** [comment]
I just had a look at this. I think our H parameters are fine. The interchange conversion is fine:

```python
In [1]: def rmin_half_to_sigma(rmin_half):
   ...:     rmin = rmin_half * 2
   ...:     return rmin / (2 ** (1/6))
   ...:

In [2]: rmin_half_to_sigma(1)
Out[2]: 1.7817974362806785
```

The OpenMM value, which feels like the weirder one, might come from from not doubling rmin_half:

```python
In [3]: def rmin_to_sigma(rmin):
   ...:     return rmin / (2 ** (1/6))
   ...:

In [4]: rmin_to_sigma(1)
Out[4]: 0.8908987181403393
```

But either way, epsilon is 0.

**Comment by j-wags** [comment]
Ahhh cool, so maybe no bug at all (or just a cosmetic bug in OMMFFs at most)

**Comment by mattwthompson** [comment]
Given that OpenMM uses the same value for hydrogens and the dummy particle, I'm going to guess that discrepancy is not intentional:

https://github.com/openmm/openmm/blob/8.0.0/wrappers/python/openmm/app/data/opc.xml#L30-L36

Taking the file Chapin cites `$AMBERHOME/dat/leap/parm/frcmod.opc` as my source of truth:

```
NONBON
  OW       1.777167268 0.2128008130
  HW          0.0000  0.0000
  EP       1.0         0.0
```

the values in the OpenMM file are clearly not accurate. Though of course this doesn't matter.

**Comment by chapincavender** [comment]
Agreed that this doesn't matter or is at worst a cosmetic bug. The AmberTools frcmod is the reference implementation of OPC according to the [publication](https://doi.org/10.1021/jz501780a), so that is the correct ground truth. I set `rmin_half` to 1.0 angstrom for both the hydrogen and virtual site, which would disagree with both the AmberTools and OpenMM implementations.

I see @mattwthompson just opened a new PR to skip the sigma comparison when epsilon is zero, and I agree that this is the best solution to avoid this kind of confusion in the future.

**Pull request merged**
